components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/validators/TokenValidationHandler.java

```diff
@@ -862,4 +869,23 @@ private void addScopesToIntrospectionResponse(OAuth2IntrospectionResponseDTO int
         String[] scopesToReturn = (String[]) ArrayUtils.addAll(validatedScopes, requestedAllowedScopes);
         introResp.setScope(OAuth2Util.buildScopeString((scopesToReturn)));
     }
+
+    private void addAudienceToIntrospectionResponse(OAuth2IntrospectionResponseDTO introResp,
+                                                      AccessTokenDO accessTokenDO) throws IdentityOAuth2Exception {
+
+        String tenantDomain = null;
+        try {
+            int appResidentTenantId = accessTokenDO.getAppResidentTenantId();
+            if (appResidentTenantId != MultitenantConstants.INVALID_TENANT_ID) {
+                tenantDomain = IdentityTenantUtil.getTenantDomain(appResidentTenantId);
+                OAuthAppDO oAuthAppDO = OAuth2Util.getAppInformationByClientId(accessTokenDO.getConsumerKey(),
```

Comment:
Is this nullable ?

Reply:
Both will be non-null values. 
Access token object is retrieved from the cache/DB.
If there is no `oAuthAppDO`, it is caught as an exception.